src/models/CallOptions.ts

```diff
@@ -0,0 +1,93 @@
+import { CallType } from './CallType';
+import { CustomMetaData } from './CustomMetaData';
+import { DTMFInput } from './DTMFInput';
+
+class CallOptions {
+  receiverCuid: string;
+  callContext: string;
+  customMetaData: CustomMetaData;
+
+  constructor(
+    receiverCuid: string,
+    callContext: string,
+    customMetaData: CustomMetaData
+  ) {
+    this.receiverCuid = receiverCuid;
+    this.callContext = callContext;
+    this.customMetaData = customMetaData;
+  }
+
+  static fromDict(dict: any) {
+    const callerCuid = dict.receiverCuid;
+    const callContext = dict.callContext;
+    const customMetaData = CustomMetaData.fromDict(dict.customMetaData);
+    const callOptions = new CallOptions(
+      callerCuid,
+      callContext,
+      customMetaData
+    );
+
+    return callOptions;
+  }
+}
+
+class M2PCallOptions extends CallOptions {
+  campaignId: string;
+  campaignEndTime: string;
+  campaignLabelList: string[];
+  dtmfInputList: DTMFInput[];
+
+  constructor(
+    receiverCuid: string,
+    callContext: string,
+    customMetaData: CustomMetaData,
+    campaignID: string,
+    campaignEndTime: string,
+    campaignLabelList: string[],
+    dtmfInputList: DTMFInput[]
+  ) {
+    super(receiverCuid, callContext, customMetaData);
+    this.campaignId = campaignID;
+
+    this.campaignEndTime = campaignEndTime;
+    this.campaignLabelList = campaignLabelList;
+    this.dtmfInputList = dtmfInputList;
+  }
+
+  static fromDict(dict: any) {
+    return new M2PCallOptions(
+      dict.receiverCuid,
+      dict.callContext,
+      dict.customMetaData,
+      dict.campaignId,
+      dict.campaignEndTime,
+      dict.campaignLabelList,
+      DTMFInput.fromList(dict.dtmfInputList)
```

Comment:
@Anush-Shand  does `DTMFInput.fromList()` handle nullability since `dict.dtmfInputList` could be nullable

Reply:
@shivamsharma2710 Currently the native android side passes an emptyList if dtmfinput list is passed as null from the core sdk and hence correctly parsed by the js side

Do we wish for it to be undefined rather than an emptylist?

Reply:
We decided go with the undefined dtmfinputList if no input is given by user while ending the call.